### PR TITLE
Add a standardized text generation sample

### PR DESCRIPTION
**Pull request by natebosch**
All languages are updating samples to show the same prompts and use
cases and adding region marks so that snippets can be automatically
extracted for the developer help sites.

Implement all common text generation samples that do not rely on file
uploads which are currently unsupported in this SDK.

### CODE REVIEW

File: samples/dart/bin/text_generation.dart [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:io';
+
+import 'package:google_generative_ai/google_generative_ai.dart';
+
+final apiKey = () {
+  final apiKey = Platform.environment['GOOGLE_API_KEY'];
```
Comment (rachelsaunders) [inline samples/dart/bin/text_generation.dart:20]: `GOOGLE_API_KEY` should be `GEMINI_API_KEY`

This comment is applicable throughout this file, this PR, and other PRs.

File: samples/dart/bin/text_generation.dart [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    stderr.writeln(r'No $GOOGLE_API_KEY environment variable');
+    exit(1);
+  }
+  return apiKey;
+}();
+
+Future<void> textGenTextOnlyPrompt() async {
+  // [START text_gen_text_only_prompt]
+  final model = GenerativeModel(
+    model: 'gemini-1.5-flash-latest',
```
Comment (rachelsaunders) [inline samples/dart/bin/text_generation.dart:31]:
```suggestion
    model: 'gemini-1.5-flash',
```

This comment is applicable throughout this file, this PR, and other PRs.

**Review by rachelsaunders — CHANGES_REQUESTED** [review]

File: samples/dart/bin/text_generation.dart (hunk 1 above)
Comment (natebosch) [reply samples/dart/bin/text_generation.dart]: Done.

File: samples/dart/bin/text_generation.dart (hunk 2 above)
Comment (natebosch) [reply samples/dart/bin/text_generation.dart]: Done.

**Review by anderdobo — APPROVED** [review]
LGTM

**Pull request merged**